attest/verifier/build.rs

```diff
@@ -86,6 +86,43 @@ fn main() {
         generate_sim_files(&data_path);
     }
 }
+
+/// An issuer of a certificate. This is the one that signs a certificate.
+struct Issuer<'a> {
+    /// The key used to sign issued certificates
+    key: SigningKey,
+    /// A description of the issuer, in the form of a Distinguished Name (DN)
+    description: &'a str,
+}
+
+/// The subject of a certificate.
+struct Subject<'a> {
+    /// The serial number of the certificate. Since this is only used in sim
+    /// environments we can limit this to 1 byte.
+    serial: u8,
+    /// The description of the subject, in the form of a Distinguished Name (DN)
+    description: &'a str,
+    /// Is this a CA certificate?
+    is_ca: bool,
+    /// The validity times of the certificate
+    validity: Validity,
+}
+
+/// Validity times
+#[derive(Debug, Clone)]
+struct Validity {
+    not_before: DateTime<Utc>,
+    not_after: DateTime<Utc>,
+}
+
+impl Validity {
+    /// Returns mbedtls times, `(not_before, not_after)`.
+    fn to_mbedtls_times(&self) -> (Time, Time) {
+        let not_before = to_mbdetls_time(self.not_before);
+        let not_after = to_mbdetls_time(self.not_after);
+        (not_before, not_after)
+    }
+}
 fn generate_sim_files(data_path: impl AsRef<Path>) {
     let data_path = data_path.as_ref();
 
```

Comment:
I wanted to clean up this function as some more sim files are coming and adding to a 150 line function didn't seem prudent.